src/apps/dolomite/arbitrum/dolomite.contract-position.template.position-fetcher.ts

```diff
@@ -0,0 +1,327 @@
+            // app tokens balances are double counted.
+            if (tokenWithMetaType.type === ContractType.APP_TOKEN) {
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              const { key, ...token } = tokenWithMetaType;
+              return { ...token, key: this.appToolkit.getPositionKey(token) };
+            }
+
+            return tokenWithMetaType;
+          }
+          return null;
+        });
+
+        const tokens = compact(maybeTokens);
+        if (maybeTokens.length !== tokens.length) return null;
+        return { address, tokens, definition };
+      }),
+    );
+
+    return Promise.all(
+      compact(skeletonsWithResolvedTokens).map(async ({ address, tokens, definition }) => {
+        const contract = multicall.wrap(this.getContract(address));
+        const baseContext = { address, contract, multicall, tokenLoader, definition };
+
+        const baseFragment: GetDataPropsParams<DolomiteMargin, DolomiteDataProps>['contractPosition'] = {
+          type: ContractType.POSITION,
+          appId: this.appId,
+          groupId: this.groupId,
+          network: this.network,
+          address,
+          tokens,
+        };
+
+        // Resolve Data Props Stage
+        const dataPropsStageParams = { ...baseContext, contractPosition: baseFragment };
+        const dataProps = await this.getDolomiteDataProps(dataPropsStageParams, underlyingTokenRequests);
+
+        // Resolve Display Props Stage
+        const displayPropsStageFragment = { ...baseFragment, dataProps };
+        const displayPropsStageParams = { ...dataPropsStageParams, contractPosition: displayPropsStageFragment };
+        const displayProps = {
+          label: await this.getLabel(displayPropsStageParams),
+          labelDetailed: await this.getLabelDetailed(displayPropsStageParams),
+          secondaryLabel: await this.getSecondaryLabel(displayPropsStageParams),
+          tertiaryLabel: await this.getTertiaryLabel(displayPropsStageParams),
+          images: await this.getImages(displayPropsStageParams),
+          statsItems: await this.getStatsItems(displayPropsStageParams),
+        };
+
+        const contractPosition = { ...baseFragment, dataProps, displayProps };
+        const key = this.appToolkit.getPositionKey(contractPosition);
+        return { ...contractPosition, key };
+      }),
+    );
+  }
+
+  async getDolomiteDataProps(
+    params: GetDataPropsParams<DolomiteMargin, DolomiteDataProps>,
+    tokenDefinitions: DolomiteTokenDefinition[],
+  ): Promise<DolomiteDataProps> {
+    const multicall = this.appToolkit.getMulticall(this.network);
+    return mapTokensToDolomiteDataProps(
+      params,
+      tokenDefinitions,
+      this.isFetchingDolomiteBalances(),
+      this.network,
+      multicall,
+    );
+  }
+
+  async getLabel(_params: GetDisplayPropsParams<DolomiteMargin>): Promise<string> {
+    return this.groupLabel;
```

Comment:
You would like to have markets currencies, like on https://zapper.xyz/apps/compound-v3?network=ethereum
USDC and ETH are labels

Reply:
I'm not sure what to change it to atm, because of how I'm envisioning this being laid out on Zapper. Fwiw, I envision three sections:

Dolomite Balances
- Displays all of our balances, which could be spread between `account.number == 0 && account.number < 100`

Borrow Positions
- Displays all of your isolated borrow positions (each has up to 32 collateral assets in it and has different health factors, amounts supplied, and amounts borrowed

Pool Positions
- This was super easy cause of the Uniswap templating.

Reply:
The label is for a position, not for a balance
Here's what will look like supply/borrow position keeping this label `= this.groupLabel` (forget about the APY and Liquidity)
<img width="815" alt="Screenshot 2023-03-29 at 3 43 13 PM" src="https://user-images.githubusercontent.com/18474228/228650316-a7270f67-0fd9-494b-8858-84f0bc853c93.png">